core/tauri/src/manager.rs

```diff
@@ -1102,12 +1102,13 @@ impl<R: Runtime> WindowManager<R> {
     S: Serialize + Clone,
     F: Fn(&Window<R>) -> bool,
   {
+    let emit_args = WindowEmitArgs::from(event, source_window_label, payload)?;
     assert_event_name_is_valid(event);
     self
       .windows_lock()
```

Comment:
One question I had here was whether it made more sense to call `windows()` instead of `windows_lock()`. This would have the effect of cloning all the windows (bad) but also of not holding the lock open. First, plausibly someone could write a filter that needs the lock, which would immediately cause a painful and unintuitive deadlock. Second, serialization can take a while, and ideally you could do other window things while that was happening.

Reply:
When triggering events we usually clone the windows, it's safer. We should change this one.

Reply:
@lucasfernog done